entity-service/src/main/kotlin/com/egm/stellio/entity/repository/QueryUtils.kt

```diff
@@ -206,17 +206,17 @@ object QueryUtils {
                 MATCH (user)
                 WHERE user.id IN ${'$'}userAndGroupIds
                 WITH user
-                MATCH (user)-[]->()-[right$authTerm]->$matchEntityClause
-                RETURN {entity: entity, right:right} as entity
+                MATCH (user)-[]->()-[right:$authTerm]->$matchEntityClause
+                RETURN {entity: entity, right: right} as entity
                 UNION
                 MATCH $matchEntityClause-[:HAS_VALUE]->(prop:Property { name: '$AUTH_PROP_SAP' })
                 WHERE prop.value IN [
                     '${AuthContextModel.SpecificAccessPolicy.AUTH_WRITE.name}',
                     '${AuthContextModel.SpecificAccessPolicy.AUTH_READ.name}'
                 ]
-                RETURN {entity:entity, right:"rCanRead"} as entity
+                RETURN entity
             }
-            WITH collect(entity) as entities, count(entity) as count
+            WITH collect(entity.entity) as entities, collect(entity.right) as rights, count(entity) as count 
```

Comment:
I think it is really error prone to manage entities and rights into 2 separate collections. I'm not even really sure you have the guarantee to iterate over the two collections in the same order?